addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1047 @@
+            eg.
+            1) if user add duration for 2 hours, return : August-23-2013 at (04-30 To 06-30) (Europe/Brussels)
+            2) if event all day ,return : AllDay, July-31-2013
+        """
+        tz = self.env.context.get('tz')
+        if not tz:  # tz can have a value False, so dont do it in the default value of get !
+            tz = self.env.context.get(self.env.user.tz)
+        tz = tools.ustr(tz).encode('utf-8')  # make safe for str{p,f}time()
+
+        format_date, format_time = self.with_context(tz=tz).get_date_formats()
+        date = fields.Datetime.context_timestamp(self.with_context(tz=self.env.user.tz), fields.Datetime.from_string(start))
+        date_deadline = fields.Datetime.context_timestamp(self.with_context(tz=self.env.user.tz), fields.Datetime.from_string(stop))
+        event_date = date.strftime(format_date)
+        display_time = date.strftime(format_time)
+
+        if zallday:
+            time = _("AllDay , %s") % (event_date)
+        elif zduration < 24:
+            duration = date + timedelta(hours=zduration)
+            time = _("%s at (%s To %s) (%s)") % (event_date, display_time, duration.strftime(format_time), tz)
+        else:
+            time = _("%s at %s To\n %s at %s (%s)") % (event_date, display_time, date_deadline.strftime(format_date), date_deadline.strftime(format_time), tz)
+        return time
+
+    def _compute_is_attendee(self):
+        for meeting in self:
+            attendee = meeting._find_my_attendee()
+            meeting.is_attendee = bool(attendee)
+
+    @api.depends('final_date')
+    def _get_rulestring(self):
+        """
+        Gets Recurrence rule string according to value type RECUR of iCalendar from the values given.
+        """
+        #read these fields as SUPERUSER because if the record is private a normal search could raise an error
+        events = self.sudo().read(['id', 'byday', 'recurrency', 'final_date', 'rrule_type', 'month_by',
+                            'interval', 'count', 'end_type', 'mo', 'tu', 'we', 'th', 'fr', 'sa',
+                            'su', 'day', 'week_list'])
+        for event in events:
+            if event['recurrency']:
+                rrule = self.compute_rule_string(event)
+                self.browse(event['id']).rrule = rrule
+            else:
+                self.browse(event['id']).rrule = ''
+
+    # retro compatibility function
+    def _rrule_write(self):
+        return self._set_rulestring(self)
+
+    @api.multi
+    def _set_rulestring(self):
+        data = self._get_empty_rrule_data()
+        if self.rrule:
+            data['recurrency'] = True
+            for event in self:
+                update_data = self._parse_rrule(self.rrule, dict(data), event.start)
+                data.update(update_data)
+                event.write(data)
+        return True
+
+    @api.model
+    def _set_date(self, values, id=False):
```

Comment:
To Improve(may be less priority): you can call this method using recorset, instead of passing id explicitly